src/lib/util.ts

```diff
@@ -50,3 +50,20 @@ export function drawBitmapToCanvas(canvas: HTMLCanvasElement, bitmap: ImageBitma
   ctx.clearRect(0, 0, canvas.width, canvas.height);
   ctx.drawImage(bitmap, 0, 0);
 }
+
+export async function canvasEncode(data: ImageData, type: string, quality?: number) {
+  const canvas = document.createElement('canvas');
+  canvas.width = data.width;
+  canvas.height = data.height;
+  const ctx = canvas.getContext('2d');
+  if (!ctx) throw Error('Canvas not initialized');
+  ctx.putImageData(data, 0, 0);
+
+  const blob = await new Promise(resolve =>
+    // TypeScript doesn't allow the promise to resolve with null for some reason.
```

Comment:
I will investigate this and PR TypeScript. I think their types are just broken.

Reply:
I think TS type inference with the revealing constructor patterns is a bit lacking. If you do `new Promise<Blob | null>(...)` it will work with `null`.